### PR TITLE
Making IC K8s 1.7 compatible and setting replicas=#workers

### PR DESCRIPTION
**Pull request by puja108**
Also set `maxUnavailable: 1` to have higher availability during updates.

Still not sure where to get the worker number from.

### CODE REVIEW

**Comment by puja108** [comment]
RFR

Thanks @kopiczko for the go template help with `len`

File: templates.go [hunk 1]
```diff
@@ -633,37 +633,28 @@ write_files:
         prometheus.io/port: '10254'
         prometheus.io/scrape: 'true'
     spec:
-      replicas: 3
+      replicas: {{len .Params.Cluster.Workers}}
```
Comment (kopiczko) [inline templates.go]: `{{.Params.Cluster.Workers | len}} is also possible. I don't know which is better.

**Comment by puja108** [comment]
ok, seems `Params` can't be accessed like that, getting:

```
cloudconfig_test.go:34: template: cloudconfig:633:29: executing "cloudconfig" at <.Params.Cluster.Work...>: can't evaluate field Params in type cloudconfig.Params
```

**Comment by kopiczko** [comment]
Skip the `.Params` part: `replicas: {{len .Cluster.Workers}}`

**Comment by puja108** [comment]
now it worked

**Comment by puja108** [comment]
@rossf7 do we still do/need vendoring dance for these changes?

**Pull request merged**